hack/externalnode/install-vm.sh

```diff
@@ -0,0 +1,174 @@
+
+check_supported_platform() {
+  echo "Checking platform supported"
+  if ! [[ $(lsb_release -rs) =~ ^(18.04|20.04)$ ]]; then
+    echoerr "Error only Ubuntu 18.04/20.04 version is supported"
+    exit 1
+  fi
+}
+
+copy_antrea_agent_files() {
+    if [[ ! -f "$CONFIG_PATH" ]]; then
+      echoerr "Error $CONFIG_PATH file not found"
+      exit 1
+    fi
+    mkdir -p $AGENT_CONF_PATH
+    echo "Copying $CONFIG_PATH to $AGENT_CONF_PATH"
+    cp $CONFIG_PATH $AGENT_CONF_PATH
+
+    if [[ ! -f "$KUBECONFIG" ]]; then
+      echoerr "Error $KUBECONFIG file not found"
+      exit 1
+    fi
+
+    echo "Copying $KUBECONFIG to $AGENT_CONF_PATH"
+    cp "$KUBECONFIG" "${AGENT_CONF_PATH}/antrea-agent.kubeconfig"
+
+    if [[ ! -f "$ANTREA_KUBECONFIG" ]]; then
+      echoerr "Error $ANTREA_KUBECONFIG file not found"
+      exit 1
+    fi
+    echo "Copying $ANTREA_KUBECONFIG to $AGENT_CONF_PATH"
+    cp "$ANTREA_KUBECONFIG" "${AGENT_CONF_PATH}/antrea-agent.antrea.kubeconfig"
+}
+
+update_antrea_agent_conf() {
+  echo "Updating clientConnection and antreaClientConnection"
+  sed -i "s|kubeconfig: |kubeconfig: $AGENT_CONF_PATH/|g" $AGENT_CONF_PATH/antrea-agent.conf
+  echo "Updating externalNodeNamespace to $AGENT_NAMESPACE"
+  sed -i "s|#externalNodeNamespace: default|externalNodeNamespace: $AGENT_NAMESPACE|g" $AGENT_CONF_PATH/antrea-agent.conf
+}
+
+start_antrea_agent_service() {
+    if [[ ! -f "$AGENT_BIN_PATH" ]]; then
+      echoerr "Error $AGENT_BIN_PATH file not found"
+      exit 1
+    fi
+    mkdir -p $AGENT_LOG_DIR
+    mkdir -p $INSTALL_PATH
+    cp "$AGENT_BIN_PATH" "$INSTALL_PATH"
+cat << EOF > /etc/systemd/system/antrea-agent.service
+Description="antrea-agent as a systemd service"
+After=network.target
+[Service]
+Environment="NODE_NAME=$NODE_NAME"
+ExecStart=$INSTALL_PATH/antrea-agent \
+--config=$AGENT_CONF_PATH/antrea-agent.conf \
+--logtostderr=false \
+--log_file=$AGENT_LOG_DIR/antrea-agent.log
+Restart=on-failure
+[Install]
+WantedBy=multi-user.target
+EOF
+    sudo systemctl daemon-reload
+    sudo systemctl enable antrea-agent
+    echo "Starting antrea-agent service"
+    sudo systemctl start antrea-agent
+    sudo systemctl status antrea-agent
+}
+
+validate_argument() {
+    if [[ $2 == --* || -z $2 ]]; then
+        echoerr "Error invalid argument for $1: <$2>"
+        print_usage
+        exit 1
+    fi
+}
+
+while [[ $# -gt 0 ]]
+do
+key="$1"
+case $key in
+    --ns)
+    AGENT_NAMESPACE="$2"
+    validate_argument $1 $2
+    shift 2
+    ;;
+    --bin)
+    AGENT_BIN_PATH="$2"
+    validate_argument $1 $2
+    shift 2
+    ;;
+    --config)
+    CONFIG_PATH="$2"
+    validate_argument $1 $2
+    shift 2
+    ;;
+    --kubeconfig)
+    KUBECONFIG="$2"
+    validate_argument $1 $2
+    shift 2
+    ;;
+    --antrea-kubeconfig)
+    ANTREA_KUBECONFIG="$2"
+    validate_argument $1 $2
+    shift 2
+    ;;
+    --nodename)
+    NODE_NAME="$2"
+    validate_argument $1, $2
+    shift 2
+    ;;
+    -h|--help)
```

Comment:
policyBypassRules is missing, although it is optional

Reply:
As this is a structure, It's not easy for the user to pass this argument. 
we could enhance the scripts in the subsequent release to take all these configurations in a file and the script can create all the required configuration files.

For the `policyBypassRules`, user has to modify the antrea-agent.conf file manually.